pkg/controller/replicaset/replica_set_test.go

```diff
@@ -1373,3 +1373,126 @@ func TestRemoveCondition(t *testing.T) {
 		}
 	}
 }
+
+func TestSlowStartBatch(t *testing.T) {
+	fn := func() error {
+		return nil
+	}
+	errFn := func() error {
+		return fmt.Errorf("fake error")
+	}
+
+	tests := []struct {
+		name                       string
+		totalBatchSize             int
+		fn                         func() error
+		expectedProcessedBatchSize int
+		expectedErr                error
+	}{
+		{
+			"totalBatchSize = 0, non-error func",
+			0,
+			fn,
+			0,
+			nil,
+		},
+		{
+			"totalBatchSize = 0, error func",
+			0,
+			errFn,
+			0,
+			nil,
+		},
+		{
+			"totalBatchSize = 10, non-error func",
+			10,
+			fn,
+			10,
+			nil,
+		},
+		{
+			"totalBatchSize = 10, error func",
+			10,
+			errFn,
+			0,
+			fmt.Errorf("fake error"),
+		},
+	}
+
+	for _, test := range tests {
+		processedBatchSize, err := slowStartBatch(test.totalBatchSize, test.fn)
+		if processedBatchSize != test.expectedProcessedBatchSize {
+			t.Errorf("%s: unexpected processed batch size, expected %d, got %d", test.name, test.expectedProcessedBatchSize, processedBatchSize)
+		}
+		if !reflect.DeepEqual(err, test.expectedErr) {
+			t.Errorf("%s: unexpected processed batch size, expected %v, got %v", test.name, test.expectedErr, err)
+		}
+	}
+}
+
+func TestGetPodsToDeleteDetails(t *testing.T) {
```

Comment:
Once we move the Sort inside getPodsToDelete, the main thing we should test is when there are `>diff` Pods and they're in different states (as described in the comment above the call to Sort).